app/src/main/kotlin/com/ark/globe/coordinates/URLParser.kt

```diff
@@ -32,6 +35,13 @@ class URLParser {
                     longitude = coordinatesString.substringAfter("/")
                 }
 
+                if(url != null && url.contains(OSM_AND_NET)) {
+                    //osmand.net link processing
+                    val coordinateString = url.substringAfter("lat=").substringBefore("&z")
+                    latitude = coordinateString.substringBefore("&lon=")
```

Comment:
It's better to convert query parameters into a mapping and retrieve values from it by keys. This would be more robust. This code can break if suddenly OSM will encode the same query parameters in the URL but in different order.